### PR TITLE
MySQL and MongoDB update resource

### PR DESCRIPTION
**Pull request by ilija-savic-devtech**
# What's the purpose of this MR?
Updating of resource
# Impact Analysis
(N/A)
# Where should the tester start?
- Run start_server.bat
- Run Postman
- When you are in Postman, go to body tab and for key values enter: name, surname, indexno, address. For values enter what you want.
- After that, go to field where you can type address and type: localhost/id, where id represents id in database
- Choose POST method and click SEND button
# How should this be manually tested?
(N/A)
# Any background context you want to provide?
(N/A)
# What are the relevant tickets?
(N/A)
# Screenshots (if appropriate)
(N/A)

### CODE REVIEW

File: database/ServiceMongoDb.php [hunk 1]
```diff
@@ -114,14 +114,31 @@ public function create()
...
     }
 
-    public function delete($id)
+    public function update($id)
     {
-        // TODO: Implement delete() method.
+        try{
+            $bulk = new BulkWrite();
+
+            $bulk->update(['_id' => intval($id)], ['$set' =>
```
Comment (goran-erhartic-devtech) [inline database/ServiceMongoDb.php]: what would happen if you wanted to edit only the address and leave other fields empty? :)

File: database/ServiceMySql.php [hunk 2]
```diff
@@ -81,14 +81,25 @@ public function create()
 		}
 	}
 
-	public function delete($id)
+	public function update($id)
 	{
+		try{
+			$sql = $this->conn->prepare("UPDATE guest.student SET name = :name, surname = :surname, indexno = :indexno, address = :address WHERE id = " . $id);
+
```
Comment (goran-erhartic-devtech) [inline database/ServiceMySql.php]: what would happen if you wanted to edit only the address and leave other fields empty? :)

Comment (ilija-savic-devtech) [reply database/ServiceMySql.php]: It will be empty fields

Comment (goran-erhartic-devtech) [reply database/ServiceMySql.php]: yes, I know, but is that a good idea?

File: public/index.php [hunk 3]
```diff
@@ -20,6 +20,10 @@
 	$crud->create();
 });
 
+$klein->respond('POST', '/[i:id]', function($request) use ($crud){
```
Comment (goran-erhartic-devtech) [inline public/index.php]: which method should we call when we want to update something?

File: database/ServiceMySql.php (hunk 2 above)
Comment (ilija-savic-devtech) [reply database/ServiceMySql.php]: I added checking of the fields, take a look

**Comment by ilija-savic-devtech** [comment]
Check it out. I made small changes in getOne and update methods.

File: database/ServiceMongoDb.php [hunk 4]
```diff
@@ -19,27 +19,57 @@ class ServiceMongoDb implements ServiceInterface
...
+        if (trim($data["indexno"]) !== "") {
+            $q['indexno'] = $data['indexno'];
+        }
+        if (trim($data["address"]) !== "") {
+            $q['address'] = $data['address'];
+        }
+        return $q;
+    }
+
+    public final function checkId($id){
```
Comment (goran-erhartic-devtech) [inline database/ServiceMongoDb.php]: public?

File: database/ServiceMySql.php [hunk 5]
```diff
@@ -8,87 +8,127 @@
...
-
-	public function update($id)
-	{
-		// TODO: Implement update() method.
-	}
+    private $conn;
 
+    private final function putValues($id, $data){
+        $putValues = array();
+        $idValues = $this->getOne($id);
```
Comment (goran-erhartic-devtech) [inline database/ServiceMySql.php]: do we really need to call getOne($id) function or?

Comment (ilija-savic-devtech) [reply database/ServiceMySql.php]: checkId($id) ?

File: database/ServiceMongoDb.php (hunk 4 above)
Comment (ilija-savic-devtech) [reply database/ServiceMongoDb.php]: I misspelled this :)

File: database/ServiceMySql.php (hunk 5 above)
Comment (goran-erhartic-devtech) [reply database/ServiceMySql.php]: yup! we only need to retrieve it from the DB and check it's values. we do not need to instantiate the obj and so on...

Comment (ilija-savic-devtech) [reply database/ServiceMySql.php]: Already done it, just wait to push

**Comment by goran-erhartic-devtech** [comment]
looks good! In next feature please implement PHPDocs for functions

**Pull request merged**